apps/mobile/src/features/approver/components/inputs-outputs-card.tsx

```diff
@@ -1,10 +1,35 @@
+import { useBtcMarketDataQuery } from '@/queries/market-data/btc-market-data.query';
 import { t } from '@lingui/macro';
 
+import { PsbtInput, PsbtOutput } from '@leather.io/bitcoin';
 import { Box, Text } from '@leather.io/ui/native';
+import { baseCurrencyAmountInQuote, createMoney } from '@leather.io/utils';
 
 import { UtxoRow } from './utxo-row';
 
-export function InputsAndOutputsCard() {
+interface InputsAndOutputsCardProps {
+  inputs: PsbtInput[];
+  outputs: PsbtOutput[];
+}
+
+export function InputsAndOutputsCard({ inputs, outputs }: InputsAndOutputsCardProps) {
+  const { data: btcMarketData } = useBtcMarketDataQuery();
+
+  function addBalance<T extends PsbtInput | PsbtOutput>(inputOutput: T) {
+    const btcBalance = createMoney(Number(inputOutput.value), 'BTC');
+    const usdBalance = btcMarketData
+      ? baseCurrencyAmountInQuote(btcBalance, btcMarketData)
+      : createMoney(0, 'USD');
```

Comment:
I've seen this used in few other places, and it can potentially cause flicker from 0 to actual balance, unless the relevant query happens to be in cache and fresh. Is there anything we're missing design-wise to avoid this?
Couple things I can think of:
1. Loading state (if applicable)
2. Making shared data like market queries globally available upon app startup

Reply:
We have been hardcoding this fallback to `0` as we had been awaiting design. That's finished now and updating this is part of https://linear.app/leather-io/issue/LEA-1726/handle-balance-loading-state-elegantly 

We need to improve this so we never show `0` unless it is actually `0` and while it is loading to show a loader and if it fails then show a fallback. [The design is here ](https://www.figma.com/design/FlmF7ropcijOb7iVSDqNn0/%F0%9F%9F%AA-App-Explorations-2024-H2?node-id=22529-339198&node-type=frame&t=QXcdDievPkzQRpoE-0). 

Once we have a static UI for it we can backfill and refactor away the `createMoney(0, 'USD')`